NitroxClient/MonoBehaviours/Gui/Chat/PlayerChat.cs

```diff
@@ -16,12 +15,15 @@ class PlayerChat : MonoBehaviour
         private GUIText chatText;
         private Coroutine timer;
         private List<String> messages;
```

Comment:
Let's do a List<ChatMessage> (new nested class) that has member variables for message and color.  You could even tuck away some of the sanitizing and line limiting logic into it.  This allows us to avoid multiple related data structures.

Reply:
If you're interested, this is called SOA (Structure Of Arrays), whereas we prefer AOS (Array Of Structures), because it groups coupled data together, and allows for neat functions that work on this data.

Sidenote: I recently looked into cInput (a Unity input library), which heavily relies on SOA. The code is extremely hard to follow, everything is crammed into a single structure, and don't even want to think about maintaining it :wink: